fml/time/dart_timestamp_provider.cc

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/time/dart_timestamp_provider.h"
+
+#include "dart_tools_api.h"
+
+namespace fml {
+
+DartTimestampProvider::DartTimestampProvider() = default;
+
+DartTimestampProvider::~DartTimestampProvider() = default;
+
+fml::TimePoint DartTimestampProvider::Now() {
+  return fml::TimePoint::FromTicks(Dart_TimelineGetTicks());
```

Comment:
Perhaps I don't understand this API but this seems off to me. Here, we are depending on the timebases of `fml::TimePoint` and `Dart_Timeline` to be identical. Looking at what was [recently exposed](https://dart-review.googlesource.com/c/sdk/+/207900/2/runtime/include/dart_tools_api.h) we'd need to take into the timebase using `Dart_TimelineGetTicksFrequency`. Maybe we are getting away with these because both are in nanoseconds?